Add org-incoming

tinloaf opened the pull request:
### Brief summary of what the package does

org-incoming is a package to ingest incoming PDFs (e.g. from scanning handwritten notes) into your org (or org-roam) files. It provides an interactive workflow of naming, dating and annotating or transcribing the files.

### Direct link to the package repository

https://github.com/tinloaf/org-incoming

### Your association with the package

I am the maintainer (and author).

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly (**Note**: There are two warnings, one "function used to take X arguments, now takes Y", and one "not known to be defined", which I assume are false positives. I'd be happy to hear how to get rid of them!)
- [x] `M-x checkdoc` is happy with my docstrings (**Note**: Sort of. It keeps complaining about lines longer than 80 characters, but all my lines are below 80 characters.)
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`org-incoming.el` with byte-compile using Emacs 28.1:
```
org-incoming.el:133:1: Warning: custom-declare-variable
    `org-incoming-parse-date-re' docstring wider than 80 characters
org-incoming.el:148:1: Warning: custom-declare-variable `org-incoming-dirs'
    docstring wider than 80 characters
org-incoming.el:167:1: Warning: custom-declare-variable
    `org-incoming-pdf-subdir' docstring wider than 80 characters
org-incoming.el:176:1: Warning: custom-declare-variable
    `org-incoming-use-roam' docstring wider than 80 characters
In org-incoming--sanitize-filename:
org-incoming.el:238:8: Warning: docstring wider than 80 characters
In org-incoming--permissive-rename-file:
org-incoming.el:255:8: Warning: docstring wider than 80 characters
In org-incoming--next-in-dir:
org-incoming.el:662:8: Warning: docstring wider than 80 characters
In end of data:
org-incoming.el:570:18: Warning: the function `org-id-get-create' is not known
    to be defined.
org-incoming.el:453:18: Warning: the function `widget-forward' is not known to
    be defined.
org-incoming.el:289:30: Warning: the function
    `org-incoming--handle-file-error' is not known to be defined.
org-incoming.el:195:18: Warning: the function `widget-value-set' is not known
    to be defined.
org-incoming.el:194:22: Warning: the function `org-read-date' is not known to
    be defined.
```

`org-incoming.el` with checkdoc 0.6.2:
```
org-incoming.el:244: Some lines are over 80 columns wide
```

`org-incoming.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- org-incoming.el#L206: You can use `not` or `null` (instead of `eq year nil`)
- org-incoming.el#L207: You can use `not` or `null`
- org-incoming.el#L208: You can use `not` or `null`
- org-incoming.el#L231: You can use `not` or `null`
- org-incoming.el#L248: You can use `not` or `null`
- org-incoming.el#L378: You can use `not` or `null`
- org-incoming.el#L382: You can use `not` or `null`
- org-incoming.el#L399: You can use `not` or `null`
- org-incoming.el#L400: You can use `not` or `null`
- org-incoming.el#L410: You can use `not` or `null`
- org-incoming.el#L533: You can use `not` or `null`
- org-incoming.el#L624: No `format` required; `error` takes an f-string
- org-incoming.el#L626: No `format` required; `error` takes an f-string
- org-incoming.el#L250: No `format` required; `message` takes an f-string
- org-incoming.el#L500: No `format` required; `message` takes an f-string
```

alphapapa commented:
This looks cool.  A couple of small suggestions about the source code: run `M-x untabify` and then enable `aggressive-indent-mode`.  You can see some of the formatting and indentation problems at, e.g. https://github.com/tinloaf/org-incoming/blob/f1ce2618207207a5c4fae4371252cf687245354e/org-incoming.el#L394

tinloaf commented:
Thanks @riscy and @alphapapa for the review (and thanks for the `aggressive-indent-mode` hint). I think I addressed all of the issues (though I needed to build a fresh Emacs to reproduce some of the compilation warnings). I also added meplazoid to the repository and it looks satisfied:  https://github.com/tinloaf/org-incoming/actions/runs/3680515563/jobs/6226202690

(@riscy : I needed to modify the workflow though - Python 3.6 seems to not be supported anymore. I'll open a PR over at melpazoid).

riscy commented:
Thank you for opening that PR on melpazoid -- hugely appreciated. :)

You've probably seen these hints, but they're not blockers:
```
- org-incoming.el#L214: You can use `not` or `null`
- org-incoming.el#L215: You can use `not` or `null`
- org-incoming.el#L216: You can use `not` or `null`
- org-incoming.el#L239: This double negation can be collapsed (`not` aliases `null`)
- org-incoming.el#L257: This double negation can be collapsed (`not` aliases `null`)
- org-incoming.el#L400: Use `when ...` instead of `unless (null ...)`
- org-incoming.el#L404: Use `when ...` instead of `unless (null ...)`
- org-incoming.el#L555: Use `when ...` instead of `unless (null ...)`
- org-incoming.el#L619: Use `when ...` instead of `unless (null ...)`
```
